app/controllers/users_controller.rb

```diff
@@ -1,4 +1,26 @@
 class UsersController < ApplicationController
   def new
+    @user = User.new
   end
+
+  def show
+    @user = User.find_by(id: params[:id])
```

Comment:
```
def show
 @user = User.find_by id: params[:id]
# Assume now @user is nil, now what will be shown on the show page
# You may think it will show nothing, but it is not a good solution
# Another thing you can think, suppose in the user's show page the @user may have some other # associated data, but if the @user is nil that came from controller will throw error in the view.  So # we need to handle this nil data in the controller
end 
```
There are several way to solve this problem depending on your application requirement. For example
```
def show 
  @user = User.find_by(id: params[:id])
  return if @user.present? 
  flash[:warning] = "User not found with the requested id"
  redirect_to root_path 
end 
```